### ISSUE TITLE
multi-line-lint is too aggressive 

### ISSUE THREAD

**Issue by waynew** [adrienverge/yamllint#356]
Consider the following:

```
foo:
  bar: >-
    this


         is just some text





        that has weird spacing but is valid
```

With this config:

```
  indentation:
    spaces: 2
    indent-sequences: true
    check-multi-line-strings: true  # or false
```
I get an error for too many blank lines. That's not correct (or pyyaml has a bug), because that parses just fine into the completely expected:

```
{'foo': {'bar': 'this\n'
                '\n'
                '\n'
                '     is just some text\n'
                '\n'
                '\n'
                '\n'
                '\n'
                '\n'
                '    that has weird spacing but is valid'}}
```

I'm running yamllint from `pre-commit` with the following config:

```
  - repo: https://github.com/adrienverge/yamllint.git
    rev: v1.25.0
    hooks:
      - id: yamllint
        files: '\.meta$'
        types: [file]
        exclude: .cicleci/*.yml
        language_version: python3
```

**Comment by adrienverge** [comment]
Hello, thanks for the report.

I understand what you mean, and this is legitimate. On the other hand I think some other users want the current behavior: report too many blank lines, *even when they are inside literals (e.g. strings)*.

A nice solution would be to add an option to the `empty-lines` rule to specify which behavior the user wants, for example `ignore-inside-literals: true|false`. However this might be tricky to implement because yamllint rules read either *lines* or *YAML tokens*, not both.

PS: If it can help, your example YAML document is 100% equivalent to this other notation:
```yaml
foo:
  bar:
    "this\n\
     \n\
     \n\
     \     is just some text\n\
     \n\
     \n\
     \n\
     \n\
     \n\
     \    that has weird spacing but is valid"
```
